frontend/src/metabase/visualizations/components/ChartSettingsWidgetList.tsx

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import _ from "underscore";
+import ChartSettingsWidget from "./ChartSettingsWidget";
+
+import {
+  ChartSettingsWidgetListHeader,
+  ChartSettingsWidgetListDivider,
+} from "./ChartSettingsWidgetList.styled";
+
+interface ChartSettingsWidgetListProps {
+  widgets: { id: string; group?: string }[];
+  extraWidgetProps: Record<string, unknown>;
+}
+
+const ChartSettingsWidgetList = ({
+  widgets,
+  extraWidgetProps,
+}: ChartSettingsWidgetListProps) => {
+  const widgetsAreGrouped = widgets.some(widget => widget.group);
+
+  if (!widgetsAreGrouped) {
+    return widgets.map(widget => (
+      <ChartSettingsWidget key={widget.id} {...widget} {...extraWidgetProps} />
+    ));
+  } else {
+    const groupedWidgets = widgets.reduce<{ [key: string]: any[] }>(
```

Comment:
Let's use `Record` from TypeScript instead. It will make this more concise.
```suggestion
    const groupedWidgets = widgets.reduce<Record<key, any[]>(
```